circuits/move/move.func.circom

```diff
@@ -136,14 +148,50 @@ template CheckRsrc(N_TL_ATRS, RSRC_IDX, PUBX_IDX, PUBY_IDX, UNOWNED, SYS_BITS) {
 
     // Moving onto enemy tile that has less resource vs what's being sent
     signal case3Logic <== BatchIsEqual(2)([
-        [tFrom[RSRC_IDX] - uFrom[RSRC_IDX], tTo[RSRC_IDX] + uTo[RSRC_IDX]],
+        [fromUpdatedTroops - uFrom[RSRC_IDX], toUpdatedTroops + uTo[RSRC_IDX]],
         [uToPub, uFromPub]]);
     signal case3LogicWrong <== NOT()(case3Logic);
     signal case3Selector <== AND()(ontoEnemy, ontoLess);
     signal case3 <== case3LogicWrong * case3Selector;
 
-    out <== BatchIsZero(7)([movedAllTroops, overflowFrom, overflowTo, 
-        fromOwnershipWrong, case1, case2, case3]);
+    out <== BatchIsZero(8)([movedAllTroops, troopUpdateIncorrect, overflowFrom, 
+        overflowTo, fromOwnershipWrong, case1, case2, case3]);
+}
+
+/*
+ * Asserts that the attacker's and defender's resources prior to moving reflect
+ * any troop updates that should have occurred.
+ */
+template CheckTroopUpdates() {
+    signal input currentTroopInterval;
+
+    signal input tFromTroops;
+    signal input tFromLastUpdate;
```

Comment:
nit: could we swap all the references to "last update" to "latest update". was confused the first time reading this